### PR TITLE
fix(ops): add node.js env variable allowlist

### PR DESCRIPTION
**Pull request by cjihrig**
This commit allows the Node compatibility layer to skip environment variable permission checks when --unstable is passed and the variable name is one that Node uses.

Fixes: https://github.com/denoland/deno/issues/15890

### CODE REVIEW

File: ext/node/lib.rs [hunk 1]
```diff
@@ -59,6 +60,36 @@ pub static NODE_GLOBAL_THIS_NAME: Lazy<String> = Lazy::new(|| {
...
+  set.insert("NODE_PENDING_PIPE_INSTANCES".to_string());
+  set.insert("NODE_PRESERVE_SYMLINKS".to_string());
+  set.insert("NODE_REDIRECT_WARNINGS".to_string());
+  set.insert("NODE_REPL_HISTORY".to_string());
+  set.insert("NODE_REPL_EXTERNAL_MODULE".to_string());
+  set.insert("NODE_SKIP_PLATFORM_CHECK".to_string());
+  set.insert("NODE_TLS_REJECT_UNAUTHORIZED".to_string());
+  set.insert("NODE_V8_COVERAGE".to_string());
+  set.insert("NO_COLOR".to_string());
+  set.insert("OPENSSL_CONF".to_string());
```
Comment (ry) [inline ext/node/lib.rs:85]: this one potentially has security implications...

File: ext/node/lib.rs [hunk 2]
```diff
@@ -59,6 +60,36 @@ pub static NODE_GLOBAL_THIS_NAME: Lazy<String> = Lazy::new(|| {
...
+  set.insert("SSL_CERT_DIR".to_string());
+  set.insert("SSL_CERT_FILE".to_string());
```
Comment (ry) [inline ext/node/lib.rs:87]: these potentially have security implications

Comment (cjihrig) [reply ext/node/lib.rs:87]: Does that mean you'd like me to remove them?

Comment (dsherret) [reply ext/node/lib.rs:87]: Maybe we should keep this list really short for now and only allow the env vars that cause a lot of prompts/pain?

Comment (cjihrig) [reply ext/node/lib.rs]: OK. I've made the list significantly smaller.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**